src/java.base/share/classes/jdk/internal/util/OperatingSystem.java

```diff
@@ -127,16 +127,25 @@ public static OperatingSystem current() {
      * Names not recognized throw ExceptionInInitializerError with IllegalArgumentException.
      */
     private static OperatingSystem initOS(String osName) {
-        // Too early to use Locale conversions, manually do uppercase
-        StringBuilder sb = new StringBuilder(osName);
-        for (int i = 0; i < sb.length(); i++) {
-            char ch = sb.charAt(i);
+        return OperatingSystem.valueOf(toUpperCase(osName));
+    }
+
+    /**
+     * {@return Simple conversion of ASCII to upper case; used when too early to use Locale conversions}
+     *
+     * @param str a string
+     * @return the string with lower case ascii converted to upper case
+     */
+    /* package-private */
+    static String toUpperCase(String str) {
```

Comment:
A bit icky to have OperatingSystem to expose a static method to convert Strings as it's nothing to do with OperatingSystem, is there anywhere else?

Reply:
Delaying the initialization of OperatingSystem made it possible to revert to String.toUpperCase(). 
Enabled by the fix in PR [14181](https://github.com/openjdk/jdk/pull/14181).